sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/ManagementChannel.java

```diff
@@ -316,10 +348,95 @@ private Message createManagementMessage(String operation, String linkName) {
         return message;
     }
 
+    /***
+     * Create a Amqp key, value map to be used to create Amqp mesage for scheduling purpose.
+     *
+     * @param messageToSchedule The message which needs to be scheduled.
+     * @return Map of key and value in Amqp format.
+     * @throws AmqpException When payload exceeded maximum message allowed size.
+     */
+    private Map<String, Object> createScheduleMessgeAmqpValue(ServiceBusMessage messageToSchedule) {
+        int maxMessageSize = MAX_MESSAGE_LENGTH_SENDER_LINK_BYTES;
+        Map<String, Object> requestBodyMap = new HashMap<>();
+        List<Message> messagesToSchedule = new ArrayList<>();
```

Comment:
1. Why are you creating a list and iterating through it for a single message?
1. You aren't taking into account the size of the send link. It is not always "MAX_MESSAGE_LENGTH_SENDER_LINK_BYTES". 
1. This doesn't need to be a method. It is only used in a single place.
1. Document that the logic is from ReactorSender, so if we bug fix one place, it can also be done in another.

Reply:
A separate method will keep `scheduleMessage` succinct and readable. I can merge them together if you still want.